Optimize group_norm_op

xiaomengy opened the pull request:
Summary: Optimize group_norm_op

Differential Revision: D14419908

BIT-silence commented:
Some benchmarks.

For N = 128, C = 256, H = 112, W = 112, group = 32 on GPU:
NCHW backward: 56.75ms -> 38.45ms
NHWC backward: 532.87ms -> 84.39ms

For N = 64, C = 256, H = 56, W = 56, group = 32 on CPU:
NCHW backward: 768.14ms -> 97.17ms
NHWC forward: 623.72 -> 92.20ms
NHWC backward: 764.38 -> 121.68ms

xiaomengy committed:
Optimize group_norm_op (#17945)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/17945

Optimize group_norm_op

Differential Revision: D14419908

fbshipit-source-id: b666b07a2c2d6f5b216bbb87aa75a7fd696fd085

houseroad reviewed (approved):
Looks good. The failed tests are unrelated and already fixed in the master.
